keel-web/src/main/kotlin/com/netflix/spinnaker/keel/apidocs/BaseModelConverter.kt

```diff
@@ -0,0 +1,57 @@
+package com.netflix.spinnaker.keel.apidocs
+
+import io.swagger.v3.core.converter.AnnotatedType
+import io.swagger.v3.core.converter.ModelConverter
+import io.swagger.v3.core.converter.ModelConverterContext
+import io.swagger.v3.core.util.RefUtils.constructRef
+import io.swagger.v3.oas.models.media.ComposedSchema
+import io.swagger.v3.oas.models.media.Schema
+
+abstract class BaseModelConverter : ModelConverter {
+
+  override fun resolve(
+    annotatedType: AnnotatedType,
+    context: ModelConverterContext,
+    chain: MutableIterator<ModelConverter>
+  ): Schema<*>? =
+    if (chain.hasNext()) {
+      chain.next().resolve(annotatedType, context, chain)
+    } else {
+      null
+    }
+
+  /**
+   * @return a reference to the schema for [T] (which must already have been defined).
+   */
+  protected inline fun <reified T> ref(): Schema<*> =
+    ref(T::class.java)
+
+  /**
+   * @return a reference to the schema for [type] (which must already have been defined).
+   */
+  protected fun ref(type: Class<*>): Schema<*> =
+    Schema<Any>().`$ref`(constructRef(type.simpleName))
```

Comment:
I hadn't come across the back-tick syntax yet. Is that a method name in this case?

Reply:
Yeah, in Java `$ref` is a valid identifier but in Kotlin it’s not. You have to back-tick anything like that. IntelliJ does it for you when you auto-complete.